src/utils/env-utils.ts

```diff
@@ -105,3 +111,49 @@ export const getEncryptionSettings = () => {
         );
     }
 };
+
+export const getGatewayAuthTokenSettings = () => {
+    const gatewayAuthTokenEndpoint =
+        process.env.REACT_APP_GATEWAY_AUTH_TOKEN_URL;
+
+    if (gatewayAuthTokenEndpoint) {
+        return { gatewayAuthTokenEndpoint };
+    } else {
+        throw new Error(
+            'Missing endpoint for creating gateway auth tokens: REACT_APP_GATEWAY_AUTH_TOKEN_URL'
+        );
+    }
+};
+
+export const storeGatewayAuthConfig = ({
+    gateway_url,
+    token,
+}: GatewayAuthTokenResponse): void => {
+    localStorage.setItem(
+        LocalStorageKeys.GATEWAY,
+        JSON.stringify({ gateway_url: gateway_url.toString(), token })
+    );
+};
+
+export const getStoredGatewayAuthConfig =
+    (): GatewayAuthTokenResponse | null => {
+        const config = localStorage.getItem(LocalStorageKeys.GATEWAY);
+
+        return config ? JSON.parse(config) : null;
+    };
+
+export const removeGatewayAuthConfig = (): void => {
+    localStorage.removeItem(LocalStorageKeys.GATEWAY);
+};
```

Comment:
Post PR: This should not be in this utils as it is not related to .env file. We should start making a local storage utils or hooks... depending on needs.